src/binding/petsc4py/src/petsc4py/PETSc/DS.pyx

```diff
@@ -16,83 +17,412 @@ cdef class DS(Object):
         self.ds  = NULL
 
     def view(self, Viewer viewer=None):
+        """View a PetscDS
+
+        Parameters
+        ----------
+        prob - the PetscDS object to view
```

Comment:
not compliant with numpydoc

Reply:
this is a draft PR, but you can review DS and AO for now